tools/bench_connect.c

```
@@ -0,0 +1,243 @@
+    struct timespec wall0, wall1;
+    clock_gettime(CLOCK_MONOTONIC, &wall0);
+
+    for (int t = 0; t < threads; t++) {
+        args[t].thread_id = t;
+        strncpy(args[t].conninfo, conninfo, sizeof(args[t].conninfo) - 1);
+        args[t].iterations = iterations;
+        args[t].warmup = warmup;
+        pthread_create(&tids[t], NULL, thread_worker, &args[t]);
+    }
+    for (int t = 0; t < threads; t++)
+        pthread_join(tids[t], NULL);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)